.github/workflows/createNewVersion.yml

```diff
@@ -37,33 +34,43 @@ jobs:
   validateActor:
     runs-on: ubuntu-latest
     outputs:
+      # TODO: fix this so we check if admin is true, as well as push
+      # https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-user-access-token
       HAS_WRITE_ACCESS: ${{ contains(fromJSON('["write", "admin"]'), steps.getUserPermissions.outputs.PERMISSION) }}
     steps:
       - name: Get user permissions
         id: getUserPermissions
-        run: echo "PERMISSION=$(gh api /repos/${{ github.repository }}/collaborators/${{ github.actor }}/permission | jq -r '.permission')" >> "$GITHUB_OUTPUT"
+        run: echo "PERMISSION=$(gh api /users/installations/${{ secrets.OS_BOTIFY_INSTALLATION_ID }}/repositories | jq -r '.repositories[].permissions')" >> "$GITHUB_OUTPUT"
         env:
-          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
 
   createNewVersion:
     runs-on: macos-latest
     needs: validateActor
-    if: ${{ fromJSON(needs.validateActor.outputs.HAS_WRITE_ACCESS) }}
+    # if: ${{ fromJSON(needs.validateActor.outputs.HAS_WRITE_ACCESS) }}
 
     outputs:
       NEW_VERSION: ${{ steps.bumpVersion.outputs.NEW_VERSION }}
 
     steps:
+      - name: Generate a token
+        id: generate_token
+        uses: tibdex/github-app-token@021a2405c7f990db57f5eae5397423dcc554159c
+        with:
+          app_id: ${{ secrets.OS_BOTIFY_APP_ID }}
+          private_key: ${{ secrets.OS_BOTIFY_PRIVATE_KEY }}
+          installation_id: ${{ secrets.OS_BOTIFY_INSTALLATION_ID }}
```

Comment:
@roryabraham this is just a draft while I work through this but what are your thoughts about having the token generation be in this step as opposed to an independent job? I was having some trouble figuring out how to securely pass the output of the token into another job because github actions redacts/blanks it out due to it being a sensitive value. If it's in the same job, I can reference it via `${{ steps.generate_token.outputs.token }}`

Reply:
So recall that `setupGitForOSBotify` is an action not a worklow, so it's always going to be scoped to just a single job (and by extension a single runner). So we shouldn't need or want to pass tokens between runners.

Does that answer your question?

Reply:
Sort of. Are you suggesting we move this step into `setupGitForOSBotify` then? And if we do that, how do we use that token in these next steps instead of the under the hood `GITHUB_TOKEN` the jobs use?

The part I think I'm missing is how we access an output from one action outside of it in another step?

Reply:
Ok, I think I figured it out and got a little further. Long story short, order of operations matters here because we're flipping between the action token used in the job and the new token for OS Botify

https://github.com/Expensify/App/actions/runs/5686198797/job/15412567692

I _think_ I'm getting this next error because I need to allow OS Botify App to override our branch protections and push to main.